### ISSUE TITLE
Aard2 use proxy for 127.0.0.1 connection

### ISSUE THREAD

**Issue by espress0** [itkach/aard2-android#36]
I can search, but I can't see it's content.

Wifi -> proxy is set
Aard make a connection to wifi's proxy: http://127.0.0.1/xxxx.....
I tried adding '127.0.0.1' to bypass list, but no success.

**Comment by itkach** [comment]
> Wifi -> proxy is set

I don't know what you mean by that.

Aard 2 starts embedded web server, typically on port 8013, to serve article content. Some software, like ad blockers on rooted devices, may block connections to this embedded web server and prevent Aard 2 from working properly. If you root your device and run software like that presumably you know what you're doing and/or are willing to accept that things may not work the way they do for other people.

I'm not sure what are you asking me to do.

**Comment by espress0** [comment]
> I don't know what you mean by that.

Step-by-step

NO-ROOTED DEVICE. (which rooting is impossible because of lacking information)
NO AD BLOCKERS
Amazon Kindle 2015

1. Go to Wi-Fi and turn it on.
2. Long-tap "mynetwork" and select "modify network".
3. Set wifi proxy here:
10.20.30.40
1234
localhost,127.0.0.1

4. Install AARD2.
5. Copy dictionary data from PC.
6. Open AARD2.
7. I can search something, but when I click a result,
request goes to Wifi's proxy server, thus I can't see nothing.

Device ---> Wifi network ---> Wifi proxy

I installed this on my device. Only this app will route the connection, but I don't understand
why AARD2 make a connection to outside instead of localhost.

http://apk-dl.com/noroot-data-firewall
(I don't want to uninstall it)

Do I make myself clear?

**Comment by espress0** [comment]
Why do you change this behavior from AARD?
In AARD(1), I can search and read it without any problems.

**Comment by espress0** [comment]
For desktop, I'm using KIWIX. Because I hate Java.

**Comment by itkach** [comment]
> Do I make myself clear?

This implies that you previously provided a thorough explanation and I didn't understand it because it's my fault and now you're annoyed. Unwarranted use of CAPS reinforces my impression that you are just being rude. So fuck you.


> http://apk-dl.com/noroot-data-firewall

That's the kind of software I was talking about, something that blocks connections. The fact that it's not targeting ads specifically or that it's not on rooted device is irrelevant.

aard2 connects to http://127.0.0.1:8013 (it may be a different port if 8013 is already taken, but typically it's 8013). You need to configure your firewall to allow this connection.

> why AARD2 make a connection to outside instead of localhost

It does not. This can be tested, for example, by going to airplane mode (on a device unencumbered by connection-blocking software) and verifying that aard2 loads articles. Articles may link to external images which are loaded only if enabled in settings.

> Why do you change this behavior from AARD?

Because aard2 is a completely different program, with different design, implementation, features and goals.


> For desktop, I'm using KIWIX. Because I hate Java.

By all means, use what works for you.

aard2-web uses Java because it uses the same implementation as aard2-android and Android uses Java. On desktop platforms there appears to be very little interest in offline wikipedia reader or a simple dictionary program without deep app integrations like aard, so I am not willing to support an entirely separate code base for desktop.


> In AARD(1), I can search and read it without any problems.
> http://apk-dl.com/noroot-data-firewall
> (I don't want to uninstall it)

To sum it up. Your issue with aard2 appears to be caused by `noroot-data-firewall` blocking  connections to http://127.0.0.1:8013. If `noroot-data-firewall` can't be configured to allow this connection then you need to take this up with developers of `noroot-data-firewall`. You may want to tone down your attitude, otherwise you are unlikely to get any help.
